### PR TITLE
Generalize and improve naming for asset interface

### PR DESCRIPTION
**Pull request by benflexcompute**
Why:
There was some hard coded stuff tailored for geometry. Removed them and also made some fixes.

### CODE REVIEW

File: flow360/component/geometry.py [hunk 1]
```diff
@@ -97,9 +97,9 @@ def _validate_geometry(self):
             if not os.path.exists(geometry_file):
                 raise Flow360FileError(f"{geometry_file} not found.")
 
-        if self.name is None and len(self.file_names) > 1:
+        if self.project_name is None and len(self.file_names) > 1:
             raise Flow360ValueError(
-                "name field is required if more than one geometry files are provided."
+                "`project_name` field is required if more than one geometry files are provided."
```
Comment (maciej-flexcompute) [inline flow360/component/geometry.py:102]: can we just use the name of first file provided? We can sent a warning

File: flow360/component/simulation/cloud.py [hunk 2]
```diff
@@ -3,20 +3,52 @@
...
+        "Project", "Folder", "Geometry", "SurfaceMesh", "VolumeMesh", "Case", "Draft"
+    ] = pd.Field(serialization_alias="sourceItemType")
+    solver_version: str = pd.Field(serialization_alias="solverVersion")
+    fork_case: bool = pd.Field(serialization_alias="forkCase")
+
+    @pd.field_validator("name", mode="after")
+    @classmethod
+    def _defautl_draft_name(cls, value):
+        if value is None:
+            return "Client " + datetime.now().strftime("%m-%d %H:%M:%S")
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/cloud.py:48]: why `Client` ?

File: flow360/component/simulation/cloud.py [hunk 3]
```diff
@@ -3,20 +3,52 @@
...
 
+def _valid_id_validator(input_id: str):
+    is_valid_uuid(input_id)
+    return input_id
+
+
+IDStringType = Annotated[str, pd.AfterValidator(_valid_id_validator)]
+
+
+class DraftPostModel(pd.BaseModel):
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/cloud.py:32]:
```suggestion
class DraftPostRequest(pd.BaseModel):
```

File: flow360/component/simulation/cloud.py (hunk 2 above)
Comment (maciej-flexcompute) [reply flow360/component/simulation/cloud.py:48]: this can be done in default_factory

File: flow360/component/simulation/cloud.py [hunk 4]
```diff
@@ -40,25 +75,23 @@ def _run(
...
+        DraftPostModel(
+            name=draft_name,
+            project_id=source_asset.project_id,
+            source_item_id=source_asset.id,
+            source_item_type=source_asset.__class__.__name__,
+            solver_version=source_asset.solver_version,
+            fork_case=fork_case,
+        ).model_dump(by_alias=True)
+    )["id"]
+
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/cloud.py:89]: interface proposition for future:
```python
class DraftDraft(ResourceDraft):
    def submit() -> Draft:
        pass
 
class Draft(Flow360Resource):
    @classmethod
    def create(name, project_id, ...):
         pass
 
    def update(simulation_json):
        pass

# and use:
draft = Draft.create(
             name=draft_name,
             project_id=source_asset.project_id,
             source_item_id=source_asset.id,
             source_item_type=source_asset.__class__.__name__,
             solver_version=source_asset.solver_version,
             fork_case=fork_case,
         ).submit()

draft_id = draft.id
...
draft.update(params.model_dump_json())
draft.webapi.post( 
 json={"upTo": target_asset.__name__, "useInHouse": True},
            method="run")
 ```

File: flow360/component/geometry.py (hunk 1 above)
Comment (benflexcompute) [reply flow360/component/geometry.py:102]: Done

File: flow360/component/simulation/cloud.py (hunk 2 above)
Comment (benflexcompute) [reply flow360/component/simulation/cloud.py:48]: Changed to default_factory and replaced Client with Draft to be consistent

File: flow360/component/simulation/cloud.py (hunk 3 above)
Comment (benflexcompute) [reply flow360/component/simulation/cloud.py:32]: Done

File: flow360/component/simulation/cloud.py [hunk 5]
```diff
@@ -1,22 +1,162 @@
...
+    @classmethod
+    def from_cloud(cls, draft_id: IDStringType) -> Draft:
+        """Load draft from cloud"""
+        return Draft(draft_id=draft_id)
+
+    def update_simulation_params(self, params: SimulationParams):
+        """update the SimulationParams of the draft"""
+
+        RestApi(DraftInterface.endpoint, id=self.id).post(
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/cloud.py:137]:
```suggestion
        self.post(
```

File: flow360/component/simulation/cloud.py [hunk 6]
```diff
@@ -1,22 +1,162 @@
...
+            json={"data": params.model_dump_json(), "type": "simulation", "version": ""},
+            method="simulation/file",
+        )
+
+    def run_up_to_target_asset(self, target_asset: type[AssetBase]) -> str:
+        """run the draft up to the target asset"""
+
+        try:
+            run_response = RestApi(DraftInterface.endpoint, id=self.id).post(
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/cloud.py:146]:
```suggestion
            run_response = self.post(
```

File: flow360/component/simulation/cloud.py (hunk 5 above)
Comment (benflexcompute) [reply flow360/component/simulation/cloud.py:137]: Done

File: flow360/component/simulation/cloud.py (hunk 6 above)
Comment (benflexcompute) [reply flow360/component/simulation/cloud.py:146]: Done

**Review by maciej-flexcompute — APPROVED** [review]

**Pull request merged**